packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -111,14 +107,29 @@ const visitors: Visitors = {
             return;
         }
 
-        const decorators = (node as DecoratatedMethodDefinition).decorators;
-        const decoratedExpression = decorators?.[0]?.expression as Expression;
+        const { decorators } = node;
+        // The real type is a subset of `Expression`, which doesn't work with the `is` validators
+        const decoratedExpression = decorators?.[0]?.expression;
         if (
             is.callExpression(decoratedExpression) &&
             is.identifier(decoratedExpression.callee) &&
             decoratedExpression.callee.name === 'wire'
         ) {
-            catalogWireAdapters(path, state);
+            // Getters and setters are methods in the AST, but treated as properties by @wire
+            // Note that this means that their implementations are ignored!
+            if (node.kind === 'get' || node.kind === 'set') {
+                const methodAsProp = b.propertyDefinition(
+                    node.key,
+                    b.identifier('undefined'),
```

Comment:
`prop = undefined` is technically not needed. We can just render `prop`.

```suggestion
                    null,
```